### PR TITLE
low hanging fruit

### PR DESCRIPTION
**Pull request by fletch3555**
Some new examples:
- digital-communication
- dutycycle-encoder
- dutycycle-input
- encoder
- flywheel-bangbang-controller
- gyro-mecanum
- hid-rumble
- quick-vision
- ramsete-controller

Also reformated/updated a few existing ones to be more consistent with java examples

### CODE REVIEW

**Comment by virtuald** [comment]
Generally we follow the Java wpilib conventions, which uses k-prefix style. I don't care that much about which it is.

File: motor-control/robot.py [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
-    a short time in the loop to allow other threads to run. This is generally
-    a good idea, especially since the joystick values are only transmitted
-    from the Driver Station once every 20ms.
+    This sample program shows how to control a motor using a joystick. In the operator control part
+    of the program, the joystick is read and the value is written to the motor.
+
+    <p>Joystick analog values range from -1 to 1 and motor controller inputs also range from -1 to 1
+    making it easy to work together.
+
+    <p>In addition, the encoder value of an encoder connected to ports 0 and 1 is consistently sent
```
Comment (auscompgeek) [inline motor-control/robot.py:9]: Javadoc HTML tags shouldn't end up in the Python examples.

```suggestion
    Joystick analog values range from -1 to 1 and motor controller inputs also range from -1 to 1
    making it easy to work together.

    In addition, the encoder value of an encoder connected to ports 0 and 1 is consistently sent
```

**Comment by fletch3555** [comment]
I'll stop adding to this so it can be properly reviewed.  All the checks are green, so should be good to go now.

File: flywheel-bangbang-controller/robot.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        # Joystick to control setpoint
+        self.joystick = wpilib.Joystick(0)
+
+        self.flywheelMotor = wpilib.PWMSparkMax(self.kMotorPort)
+        self.encoder = wpilib.Encoder(self.kEncoderAChannel, self.kEncoderBChannel)
+
+        self.bangBangControler = wpimath.controller.BangBangController()
+
+        # Simulation classes help us simulate our robot
+        self.flywheelSim = wpilib.simulation.FlywheelSim(
```
Comment (virtuald) [inline flywheel-bangbang-controller/robot.py:62]: This should go in physics.py

File: flywheel-bangbang-controller/robot.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        )
+
+        # Controls a motor with the output of the BangBang controller and a
+        # feedforward. The feedforward is reduced slightly to avoid overspeeding
+        # the shooter.
+        self.flywheelMotor.setVoltage(
+            bangOutput + 0.9 * self.feedforward.calculate(setpoint)
+        )
+
+    def _simulationPeriodic(self):
```
Comment (virtuald) [inline flywheel-bangbang-controller/robot.py:94]: This does work, but I'd rather rework this example to use a separate physics.py instead.

File: ramsete-controller/robot.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import wpimath.geometry
+import wpimath.trajectory
+import wpimath.units
+
+from drivetrain import Drivetrain
+
+
+class MyRobot(wpilib.TimedRobot):
+    """
+    Uses the CameraServer class to automatically capture video from a USB webcam and send it to the
```
Comment (virtuald) [inline ramsete-controller/robot.py:20]: wat

File: ramsete-controller/robot.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+        # Create and push Field2d to SmartDashboard.
+        wpilib.SmartDashboard.putData(self.field)
+
+        # Push the trajectory to Field2d.
+        self.field.getObject("traj").setTrajectory(self.trajectory)
+
+    def autonomousInit(self):
+        # Initialize the timer.
+        self.timer = wpilib.Timer()
```
Comment (virtuald) [inline ramsete-controller/robot.py:63]: Don't need to recreate the timer here.

**Review by virtuald — CHANGES_REQUESTED** [review]
Looks like there are conflicts because of your file header changes.

File: ramsete-controller/robot.py (hunk 4 above)
Comment (fletch3555) [reply ramsete-controller/robot.py:20]: What? Doesn't everyone use CameraServer in their ramsete implementation...? _sigh_... copy/paste got me again...

File: ramsete-controller/robot.py (hunk 5 above)
Comment (fletch3555) [reply ramsete-controller/robot.py:63]: I believe the java example uses that as a way to re-initialize the timer.  Should I replace it with `timer.reset()` or just remove the line entirely?

File: flywheel-bangbang-controller/robot.py (hunk 3 above)
Comment (fletch3555) [reply flywheel-bangbang-controller/robot.py]: I haven't done anything with pyfrc's physics sim stuff before, so I may have done something wrong, but it's something.  I did run sim mode and confirm it seems to spin up/down the motor correctly (as best I can tell with a simple visual inspection at least)

Let me know how that looks.

Comment (virtuald) [reply flywheel-bangbang-controller/robot.py]: Looks good to me.

**Pull request merged**